ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTPayload.scala

```diff
@@ -198,32 +228,50 @@ object AuthServiceJWTCodec {
   // ------------------------------------------------------------------------------------------------------------------
   // Decoding
   // ------------------------------------------------------------------------------------------------------------------
-  def readFromString(value: String): Try[AuthServiceJWTPayload] = {
-    for {
-      json <- Try(value.parseJson)
-      parsed <- Try(readPayload(json))
-    } yield parsed
+  def readAudienceBasedToken(value: JsValue): AuthServiceJWTPayload = value match {
```

Comment:
Rather than having `readAudienceBasedToken` and `readScopeBasedToken` can we not have one `readToken(value: JsValue, format:  StandardJWTTokenFormat)` ?